doc/languages-frameworks/javascript.section.md

```diff
@@ -375,6 +373,37 @@ Assuming the following directory structure, we can define `sourceRoot` and `pnpm
   pnpmRoot = "frontend";
 ```
 
+#### PNPM Workspaces
+
+If you need to use a PNPM workspace for your project, then set `pnpmWorkspace = "<workspace project name>"` in your `pnpm.fetchDeps` call,
+which will make PNPM only install dependencies for that workspace package.
+
+For example:
+
+````nix
+...
+pnpmDeps = pnpm.fetchDeps {
+  ...
+  pnpmWorkspace = "@astrojs/language-server";
+}
+
+The above would make the fetchDeps call only install dependencies for the `@astrojs/language-server` workspace package.
+Note that you do not to set `sourceRoot` to make this work.
+
+#### Additional PNPM Commands
+If you require setting an additional PNPM configuration setting(such as `dedupe-peer-dependents` or similar),
+set `extraPnpmCommands` to the right commands to run. For example:
+```nix
+...
+pnpmDeps = pnpm.fetchDeps {
+  ...
+  extraPnpmCommands = ''
```

Comment:
I'd name this `prePnpmInstall`, to make it similar to the `preInstall` hook, which can't be used in this example.

Another option, which can be implemented in addition is an interface like this:

```nix
# ...
pnpmDeps = pnpm.fetchDeps {
  # ...
  extraPnpmConfig = {
    dedupe-peer-dependants = "false";
  };
};
```

This should be easily implemented via something like this (not tested):

```nix
let 
  prePnpmInstall = lib.pipe [
    (lib.mapAttrsToList (n:v: "pnpm config set ${n} ${builtins.toString v}"))
    (lib.concatStringsSep "\n")
  ] extraConfig;
in
  stdenvNoCC.mkDerivation # ...
  # ...
    installPhase = ''
      # ...
    '';
  # ...
```

You could also combine both interfaces by allowing both a string `prePnpmInstall` and a declerative `extraPnpmConfig` attribute set.

Reply:
> Another option, which can be implemented in addition is an interface like this:

After playing with this interface a bit I reached the conclusion that it will be a bit confusing to have one declarative argument that only fits the `pnpm.fetchDeps` arguments, and that can't be used as an attribute set argument to the `mkDerivation` builder - because it is an attribute set.

Hence I renamed the string option to `prePnpmInstall` and I'm marking this conversation as resolved.